### PR TITLE
Pass user specified `cluster_dims` to runtime

### PR DESCRIPTION
**Pull request by whitneywhtsang**

### CODE REVIEW

**Comment by whitneywhtsang** [comment]
It is still unclear how exactly Triton runtime should use `cluster_dims` for quantum dispatch, but this PR passes user specified `cluster_dims` to Triton runtime.

File: third_party/intel/include/Dialect/TritonIntelGPU/Transforms/Passes.h [hunk 1]
```diff
@@ -15,6 +15,14 @@
 
 namespace mlir::triton::gpu::intel {
 
+// Used by Triton runtime
+struct ClusterInfo {
+  ClusterInfo() : clusterDimX(1), clusterDimY(1), clusterDimZ(1) {}
+  int clusterDimX;
+  int clusterDimY;
+  int clusterDimZ;
+};
```
Comment (etiotto) [inline third_party/intel/include/Dialect/TritonIntelGPU/Transforms/Passes.h:24]:
```suggestion
struct ClusterInfo {
  ClusterInfo() = default;
  unsigned dimX = 1u;
  unsigned dimY = 1u;
  unsigned dimZ = 1u;
};
```

File: third_party/intel/triton_xpu.cc [hunk 2]
```diff
@@ -84,6 +84,19 @@ void init_triton_intel(py::module &&m) {
   init_triton_intel_passes_ttir(passes.def_submodule("ttir"));
   init_triton_intel_passes_ttgpuir(passes.def_submodule("ttgpuir"));
 
+  // cluster info
+  py::class_<gpu::intel::ClusterInfo>(m, "ClusterInfo")
+      .def(py::init<>())
+      .def_readwrite("clusterDimX", &gpu::intel::ClusterInfo::clusterDimX)
```
Comment (etiotto) [inline third_party/intel/triton_xpu.cc:90]: clusterDimX --> dimX etc ...

Comment (whitneywhtsang) [reply third_party/intel/triton_xpu.cc:90]: would like to keep it as the same as Nvidia if you don't feel strongly.

Comment (etiotto) [reply third_party/intel/triton_xpu.cc:90]: OK I didn't realize NVIDIA uses that name. Sure.

File: third_party/intel/include/Dialect/TritonIntelGPU/Transforms/Passes.h (hunk 1 above)
Comment (whitneywhtsang) [reply third_party/intel/include/Dialect/TritonIntelGPU/Transforms/Passes.h]: Done.

**Review by etiotto — APPROVED** [review]

**Pull request merged**